src/monitoring.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use lazy_static::lazy_static;
+use prometheus::{register_int_gauge_vec, IntGaugeVec};
```

Comment:
I think official prom library is https://github.com/prometheus/client_rust which appears to be better in most ways (see linked comparison issue in the readme)

Reply:
yes, that's an option.
I'm new to rust, and `prometheus` which is widely used projects like ``tikv`.
I'm ok to switch if you think the official one is better.

Reply:
we could also consider https://docs.rs/opentelemetry-prometheus/latest/opentelemetry_prometheus/ since it could allow us to support other Otel exporter types (OTLP)... but i suspect the UX will be bad and the performance worse

Reply:
anyhow, I think tikv made the `prometheus` one since there was none. Later prometheus community made an official implementation that is better